cabi/src/proguard.rs

```diff
@@ -77,3 +77,129 @@ ffi_fn! {
         Ok(SymbolicProguardMappingView::as_rust(view).has_line_info())
     }
 }
+
+use proguard::{ProguardMapper, ProguardMapping, StackFrame};
+
+/// Represents a Java Stack Frame.
+#[repr(C)]
+pub struct SymbolicJavaStackFrame {
+    pub klass: SymbolicStr,
+    pub method: SymbolicStr,
+    pub file: SymbolicStr,
+    pub line: usize,
+}
+
+/// The result of remapping a Stack Frame.
+#[repr(C)]
+pub struct SymbolicProguardRemapResult {
+    pub frames: *mut SymbolicJavaStackFrame,
+    pub len: usize,
+}
+
+pub struct OwnedProguardMapper<'s> {
+    _source: String,
+    mapping: ProguardMapping<'s>,
+    mapper: ProguardMapper<'s>,
+}
+
+/// Represents a proguard mapper.
+pub struct SymbolicProguardMapper;
+
+impl ForeignObject for SymbolicProguardMapper {
+    type RustObject = OwnedProguardMapper<'static>;
+}
+
+ffi_fn! {
+    /// Creates a proguard mapping view from a path.
+    unsafe fn symbolic_proguardmapper_open(
+        path: *const c_char
+    ) -> Result<*mut SymbolicProguardMapper> {
+        let path = CStr::from_ptr(path).to_str()?;
+        let source = std::fs::read_to_string(path)?;
+        let mapping = ProguardMapping::new(std::mem::transmute(source.as_str()));
```

Comment:
This `transmute` hides a lifetime cast. Would you consider to use [`SelfCell`](https://docs.rs/symbolic/7.3.3/symbolic/common/struct.SelfCell.html) for this? Otherwise, it would be good to add a few comments about safety and lifetimes here, since this is quite easy to miss.

I believe this *should* be safe, though, since a constant `String` has a stable memory address.